catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/DurableFileSystemFileConsumer.java

```diff
@@ -38,36 +44,67 @@
   @Override
   protected boolean doPoll(String sha1) {
     if (observer != null) {
-      observer.addListener(listener);
+      observer.setListener(listener);
       observer.checkAndNotify();
-      observer.removeListener(listener);
-      fileSystemPersistenceProvider.store(sha1, observer);
+      observer.removeListener();
       return true;
     } else {
       return isMatched(null, null, null);
     }
   }
 
   @Override
-  protected void initialize(String fileName, String sha1) {
+  protected void initialize(String fileName) {
     if (fileSystemPersistenceProvider == null) {
       fileSystemPersistenceProvider = new FileSystemPersistenceProvider(getClass().getSimpleName());
     }
+    if (jsonSerializer == null) {
+      jsonSerializer = new JsonPersistantStore(getClass().getSimpleName());
+    }
+
     if (observer == null && fileName != null) {
-      if (fileSystemPersistenceProvider.loadAllKeys().contains(sha1)) {
-        observer = (FileAlterationObserver) fileSystemPersistenceProvider.loadFromPersistence(sha1);
+
+      observer = AsyncFileAlterationObserver.load(new File(fileName), jsonSerializer);
+
+      //  Backwards Compatibility
+      if (observer == null && isOldVersion(fileName)) {
+        observer = backwardsCompatibility(fileName);
       } else {
-        observer = new FileAlterationObserver(new File(fileName));
+        observer = new AsyncFileAlterationObserver(new File(fileName), jsonSerializer);
       }
     }
   }
 
+  private boolean isOldVersion(String fileName) {
+    String sha1 = DigestUtils.sha1Hex(fileName);
+    return fileSystemPersistenceProvider.loadAllKeys().contains(sha1);
+  }
+
+  private AsyncFileAlterationObserver backwardsCompatibility(String fileName) {
+
+    String sha1 = DigestUtils.sha1Hex(fileName);
+    AsyncFileAlterationObserver newObserver =
```

Comment:
❓Shouldn't we use `AsyncFileAlterationObserver.load()` here instead?

Reply:
No because it's a new instance, not a instance that is file backed